### DISCUSSION TITLE
Response data return only interger or string

### DISCUSSION THREAD

**Discussion by binhvq** [fastapi/fastapi#6461] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import typing
from datetime import datetime
from starlette.responses import JSONResponse
import json


class CustomJSONResponse(JSONResponse):
    def render(self, content: typing.Any) -> bytes:
        print(content, type(content['now']))
        # {'now': '2021-12-07T11:43:41.019680'} <class 'str'>
        return json.dumps(content).encode("utf-8")


app = FastAPI(docs_url="/bdocs", redoc_url=None, default_response_class=CustomJSONResponse)


@app.get('/')
async def index():
    return {'now': datetime.now()}
```

### Description

Data datetime converted to str before json encode

### Operating System

Linux

### FastAPI Version

0.70.0

### Python Version

Python 3.9.4

### Additional Context

result on response is datetime but custom response with `default_response_class` received is string.

**Comment by panla** [comment]
datetime is not a format/type in JSON,

**Comment by binhvq** [comment]
> datetime is not a format/type in JSON,
Yes, i want custom response object from datetime auto convert to unix timestamp on some router with depends.

**Comment by panla** [comment]
can you try it?

```python
class Encoder(json.JSONEncoder):

    def default(self, obj):
        if isinstance(obj, datetime):
            return obj.strftime('%Y-%d-%m %H:%M:%S')
        elif str(obj).lower() == 'nan':
            return None
        else:
            return obj.__str__()

json.dumps(content, cls=Encoder, ensure_ascii=False)
```

**Comment by oligond** [comment]
Related to #2117.

A minimal workaround to get your intended result is to return an instance of `CustomJSONResponse` instead of the dict:

```python
@app.get('/')
async def index():
    return CustomJSONResponse(content={'now': datetime.now()})
```

You might also need to apply some more modifications (setting status code, mime type of the response), see the issue referenced above.

**Comment by binhvq** [comment, marked as answer]
This worked for me
```
pydantic.json.ENCODERS_BY_TYPE[datetime] = lambda dt: dt.replace(tzinfo=pytz.utc).isoformat()
```
